File: mobile/lib/features/detection/growth_stage_controller.dart
import '../../core/image/picked_image.dart';
import '../../models/growth_stage.dart';
import '../../services/growth_stage_api.dart';
import 'detection_controller.dart';

/// Crop-growth-stage detection controller bound to
/// `POST /api/growth-stage/detect`.
class GrowthStageController extends DetectionController<GrowthStageResult> {
  GrowthStageController(GrowthStageApi api, ImagePickerFn pickImage)
      : super(
          runner: ({
            required List<int> imageBytes,
            required String imageName,
            String? cropName,
          }) =>
              api.detect(
                imageBytes: imageBytes,
                imageName: imageName,
                cropName: cropName,
              ),
          pickImage: pickImage,
        );
}

File: config/core/controllers/growth_stage_controller.py
"""
KisanAI OS
Crop Growth Stage Detection Controller

Thin controller for the AI crop-growth-stage detection feature.
"""

from config.core.services.growth_stage_service import GrowthStageService


class GrowthStageController:
    """Crop Growth Stage Detection Controller"""

    def __init__(self, service=None):
        self.service = service or GrowthStageService()

    def detect(self, file, crop_name=None):
        return self.service.detect(file, crop_name)
